service/src/main/java/bio/terra/drshub/services/RestTemplateFactory.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.drshub.services;
+
+import bio.terra.drshub.config.MTlsConfig;
+import lombok.extern.slf4j.Slf4j;
+import nl.altindag.ssl.SSLFactory;
+import nl.altindag.ssl.util.Apache4SslUtils;
+import nl.altindag.ssl.util.PemUtils;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@Slf4j
+public class RestTemplateFactory {
+
+  private static final Integer CONNECTION_POOL_SIZE = 500;
```

Comment:
For an unboxed value, it's better to use an unboxed type. This makes it explicit to the reader (and the compiler) that the value can never contain `null`.
```suggestion
  private static final int CONNECTION_POOL_SIZE = 500;
```